src/ngResource/resource.js

```diff
@@ -656,6 +663,14 @@ angular.module('ngResource', ['ng']).
             if (!isInstanceCall && cancellable) {
               timeoutDeferred = $q.defer();
               httpConfig.timeout = timeoutDeferred.promise;
+
+              if(simulatedTimeout) {
+                simulatedTimeoutPromise = $timeout(function() {
+                  if(timeoutDeferred) {
+                    timeoutDeferred.resolve();
+                  }
+                }, simulatedTimeout);
```

Comment:
Yes, but I wanted to be sure, timeoutDeferred is not null already, so extra if() here. But may be it's obvious.